compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -124,6 +124,9 @@ use self::traits::check_trait_impl_method_matches_declaration;
 pub(crate) use path_resolution::{TypedPath, TypedPathSegment};
 pub use primitive_types::PrimitiveType;
 
+/// Maximum number of items on the interpreter stack, limiting recursion during comptime.
+pub const MAX_INTERPRETER_CALL_STACK_SIZE: usize = 100;
```

Comment:
```suggestion
pub const MAX_INTERPRETER_CALL_STACK_SIZE: usize = 1000;
```
Might be better to match what we do for ACIR

Reply:
Perhaps we should do an earlier pass and point out any expression which is too deep to evaluate?

Reply:
Or maybe at this point rather than a new pass or expanding the depth checks we should just look into making the interpreter iterative and the workload that would require.

Reply:
I'm not sure that would help us in this case: with your example we never enter `Elaborator::elaborate_comptime_block`, I think it dies before that already.

Reply:
I'm not necessarily saying a new stand-alone pass should be added, just that whatever traverses the AST first, maybe the `Elaborator`, should be aware how deep it is at any moment into processing an expression, emit an error pointing at the expression which exceeds the limits, and... not sure what it can do next.

Reply:
> I'm not necessarily saying a new stand-alone pass should be added, just that whatever traverses the AST first, maybe the `Elaborator`, should be aware how deep it is at any moment into processing an expression, emit an error pointing at the expression which exceeds the limits, and... not sure what it can do next.

Ah yes ok. The `Elaborator` may be the best place for this kind of detection. It follows the methodology of https://github.com/noir-lang/noir/issues/10727 where we should not code that we know will fail evaluation.

Reply:
Created https://github.com/noir-lang/noir/issues/10948 to capture this.